File: prob5.rb
#!/usr/bin/env ruby
def div_by_all n 
  if n%10==0 and n%11==0 and n%12==0 and n%13==0 and n%14==0 and n%15==0 and n%16==0 and n%17==0 and n%18==0 and n%19==0 and n%20==0 then
    return true
  end
  return nil
end

def is_div_all n 
  if n%10==0 and n%11==0 and n%12==0 and n%13==0 and n%14==0 and n%15==0 and n%16==0 and n%17==0 and n%18==0 and n%19==0 and n%20==0  then
    printf "#{n} works!\n"
  else 
    printf "no way man\n"
  end
end

i = 2520
until div_by_all(i) do
  i += 2520
end
is_div_all i
print "found it: #{i}\n"

File: prob5.c
#include<stdio.h>

int div_by_all(long n) {
       if (n%10>0 || n%11>0 || n%12>0 || n%13>0 || n%14>0 || n%15>0 || n%16>0 || n%17>0 || n%18>0 || n%19>0 || n%20>0) {   
         return 0;
       } else {
         return 1;
       }
}

int main() {  
  long i = 100L;
  while (!div_by_all(i)) {
    i += 30L;
  }
  is_div_all(i);
  printf("found it: %ld\n", i);
}

int is_div_all (long n) {
  if (n%10<1 && n%11<1 && n%12<1 && n%13<1 && n%14<1 && n%15<1 && n%16<1 && n%17<1 && n%18<1 && n%19<1 && n%20<1) {    
    printf("%ld works!\n", n);
  }
  else {
    printf("no way man\n");
  }
  return 1;
}
